### ISSUE TITLE
TNS Preview doesn't work on iPhone (real device)

### ISSUE THREAD

**Issue by pavel-arapov** [NativeScript/nativescript-cli#5117]
**Environment**
Provide version numbers for the following components (information can be retrieved by running `tns info` in your project folder or by inspecting the `package.json` of the project):
 - CLI: 6.1.2
 - iOS Runtime: 13.1.3, iPhone 11 Pro (real device)

**Describe the bug**
Preview App on iOS doesn't show any application only the default app with the list of elements. I've tried to run the application from CLI (tns preview), online playground - same result. Re-installing of Playground & Preview applications didn't help. Please, advice. Sorry if it is not the right place to discuss the issue.

**Comment by rosen-vladimirov** [comment]
Hey @pavel-arapov , 
Can you share some more details about your project? Did it work before updating the preview application? Can you send us a saved session from the playground site and the package.json of the application you try to use with `tns preview`?

**Comment by pavel-arapov** [comment]
Hello @rosen-vladimirov ,
I try to run any project, for example, https://market.nativescript.org/plugins/tns-template-tab-navigation-vue - this one from CLI. It works perfectly in Simulator, but I cannot see it on a real device.

**Comment by rosen-vladimirov** [comment]
Hey @pavel-arapov ,
Can you please try the following steps:
1. Update your NativeScript CLI to latest version: `npm install -g nativescript`
2. Create a new application: `tns create my-app-name --template tns-template-tab-navigation-vue`
3. Try `tns preview` in the newly created app and see if it works on device.

NOTE: Once you scan the code, the device will open the Preview application, you may see a white screen for a while and then the application will be restarted. At this point you have to see the application you've just scanned.

If this does not help, can you please send us information about the version of the Preview application on your device.

**Comment by pavel-arapov** [comment]
Incredible! After the upgrade of TNS the to version 6.2.0 it works perfectly with newly recreated apps. It doesn't work with the old ones. I'm not going to investigate the reason, probably it is related to the differs version between the local plugins installed and plugins in Preview. Thank you for your help.

**Comment by pavel-arapov** [comment]
Unfortunately, I was mistaken about the "old" apps, actually, I encounter the following error now on every application I try to run.

Model | OS version | Preview app version | Runtime version
 -- | -- | -- | --
 iOS device | iOS 13.1.3 | 1.26.0 | 6.2.0



```LOG from device Dead11Pro: CONSOLE INFO file:///app/vendor.js:145:32: HMR: Hot Module Replacement Enabled. Waiting for signal.
2019-11-07 12:10:35.930 nsplaydev[4173:4561676] PlayLiveSync: applicationDidFinishLaunching
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(NativeRadSideDrawer)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativestacklayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativegridlayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativestacklayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativestacklayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativestacklayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativestacklayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativestacklayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativescrollview)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativestacklayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativegridlayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativestacklayout), ElementNode(nativegridlayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativegridlayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativestacklayout), ElementNode(nativegridlayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativegridlayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativestacklayout), ElementNode(nativegridlayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativegridlayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativestacklayout), ElementNode(nativegridlayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativestacklayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativestacklayout), ElementNode(nativestacklayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativegridlayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativestacklayout), ElementNode(nativegridlayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativescrollview), ElementNode(nativestacklayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativescrollview))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativestacklayout), ElementNode(nativegridlayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativeradsidedrawer), ElementNode(nativestacklayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(NativeFrame)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(NativePage)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(NativeActionBar)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativenavigationbutton)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativeactionbar), ElementNode(nativenavigationbutton))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativeactionitem)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativeactionbar), ElementNode(nativeactionitem))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativeactionbar), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativepage), ElementNode(nativeactionbar))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativegridlayout)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> CreateElement(nativelabel)\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativegridlayout), ElementNode(nativelabel))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativepage), ElementNode(nativegridlayout))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativeframe), ElementNode(nativepage))\''
CONSOLE LOG file:///app/tns_modules/nativescript-vue/dist/index.js:10140:14: '\'{NSVue (Vue: 2.6.10 | NSVue: 2.4.0)} -> AppendChild(ElementNode(nativeradsidedrawer), ElementNode(nativeframe))\''
2019-11-07 12:10:36.064 nsplaydev[4173:4561676] PlayLiveSync: openURL
2019-11-07 12:10:36.064 nsplaydev[4173:4561676] PlayLiveSync: Starting application using url nsplay://boot?instanceId=iP3VrV9Qj&pKey=pub-c-d7893276-cc78-4d18-8ab0-becba06e43de&sKey=sub-c-3dad1ebe-aaa3-11e8-8027-363023237e0b&template=play-ng&hmr=1
2019-11-07 12:10:36:071 nsplaydev[4173:4561676] <Logger::com.pubnub.pubnub-objc> File logger enabled.
2019-11-07 12:10:36:071 nsplaydev[4173:4561676] <Logger::com.pubnub.pubnub-objc> Log files stored in: /var/mobile/Containers/Data/Application/9DBB17A8-EA13-4728-9027-9B5B27F99100/Documents/Logs/com.pubnub.pubnub-objc.
2019-11-07 12:10:36:071 nsplaydev[4173:4561676] <Logger::com.pubnub.pubnub-objc> Disk quota changed to 52428800 bytes.
2019-11-07 12:10:36.080 nsplaydev[4173:4561676] PlayLiveSync: Applying filter expression - (platform == 'ios' || platform == 'all') && (msvi <= 39 || di == 'FF399D75-F813-485A-A9DF-0B99AB4FF60D')
2019-11-07 12:10:36.080 nsplaydev[4173:4561676] PlayLiveSync: PubNub listener started and subscribed to channel m-ch-iP3VrV9Qj
CONSOLE ERROR file:///app/tns_modules/@nativescript/core/trace/trace.js:174:22: Error: Could not load action bar icon: res://navigation/menu
2019-11-07 12:10:36.112 nsplaydev[4173:4561676] PlayLiveSync: Uncaught Exception
2019-11-07 12:10:36.113 nsplaydev[4173:4561676] PlayLiveSync: Sending crash report
2019-11-07 12:10:40.118 nsplaydev[4173:4561676] *** JavaScript call stack:
(
	0   addObject@[native code]
	1   populateMenuItems@file:///app/tns_modules/@nativescript/core/ui/action-bar/action-bar.js:224:31
	2   update@file:///app/tns_modules/@nativescript/core/ui/action-bar/action-bar.js:210:27
	3   onLoaded@file:///app/tns_modules/@nativescript/core/ui/page/page.js:249:28
	4   @file:///app/tns_modules/@nativescript/core/ui/core/view-base/view-base.js:312:93
	5   callFunctionWithSuper@file:///app/tns_modules/@nativescript/core/ui/core/view-base/view-base.js:305:9
	6   callLoaded@file:///app/tns_modules/@nativescript/core/ui/core/view-base/view-base.js:312:31
	7   loadView@file:///app/tns_modules/@nativescript/core/ui/core/view-base/view-base.js:440:22
	8   _addViewCore@file:///app/tns_modules/@nativescript/core/ui/core/view-base/view-base.js:435:20
	9   _addView@file:///app/tns_modules/@nativescript/core/ui/core/view-base/view-base.js:421:22
	10  viewWillAppear@file:///app/tns_modules/@nativescript/core/ui/page/page.js:81:23
	11  UIApplicationMain@[native code]
	12  _start@file:///app/tns_modules/@nativescript/core/application/application.js:324:22
	13  run@file:///app/tns_modules/@nativescript/core/application/application.js:366:9
	14  $start@file:///app/tns_modules/nativescript-vue/dist/index.js:14050:18
	15  @file:///app/bundle.js:1328:10
	16  ./app.js@file:///app/bundle.js:1332:34
	17  __webpack_require__@file:///app/runtime.js:751:34
	18  checkDeferredModules@file:///app/runtime.js:44:42
	19  webpackJsonpCallback@file:///app/runtime.js:31:39
	20  anonymous@file:///app/bundle.js:2:61
	21  evaluate@[native code]
	22  moduleEvaluation@:1:11
	23  promiseReactionJob@:1:11
)
2019-11-07 12:10:40.118 nsplaydev[4173:4561676] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil'
*** First throw call stack:
(0x18cd91c30 0x18caac0c8 0x18cdea5f4 0x18cde9ba8 0x18cc6266c 0x1036b4044 0x1036b05c8 0x1036b00c4 0x102b84eb0 0x1036af2c8 0x1036ad644 0x1036ad644 0x1036ad644 0x1036ad644 0x1036ad644 0x1036ad5a4 0x1036ad5a4 0x1036ad644 0x1036ad5a4 0x1036ad644 0x1036990ec 0x10333414c 0x102bf8380 0x102bf7eac 0x102bf856c 0x1036b0ea4 0x1036b41b4 0x1907f6854 0x1907f6ef4 0x1907601c0 0x190760db4 0x1907620f8 0x1907454e0 0x1912cfabc 0x18caa7af0 0x19386e0f4 0x19386e3fc 0x193881964 0x1937c6c1c 0x1937f1bd8 0x19372b33c 0x1937f9d30 0x18cce2f34 0x18cd0d91c 0x18cd0cfe8 0x18cd07c20 0x18cd07098 0x196e71534 0x190e277ac 0x1036b4044 0x1036b05c8 0x1036b00c4 0x102b84eb0 0x1036af2c8 0x1036ad644 0x1036ad644 0x1036ad644 0x1036ad644 0x1036ad644 0x1036ad644 0x1036ad644 0x1036ad5a4 0x1036ad644 0x1036990ec 0x10333414c 0x1035027a8 0x102b9d4c0 0x102d01728 0x1036af02c 0x1036ad644 0x1036ad644 0x1036990ec 0x10333414c 0x103502974 0x102cfaec4 0x102b93eb0 0x102c0682c 0x103d9e25c 0x1023e3524 0x18cb86f30)
libc++abi.dylib: terminating with uncaught exception of type NSException
2019-11-07 12:10:40.118 nsplaydev[4173:4561676] PlayLiveSync: Uncaught Exception
```

**Comment by rosen-vladimirov** [comment]
Hey @pavel-arapov ,
Thank you for the detailed information. We've investigated this case and here's what we've found. In the latest Preview application, some of the resources have been moved to a new location, but they have not been moved in the templates. We'll provide a fix in our templates, but for your existing applications, that you would like to use in Preview app, you will have to replace the `res://navigation/menu` with `res://menu`. 
Can you give it a try and inform us if this resolves the issue on your side.

**Comment by pavel-arapov** [comment]
Hi @rosen-vladimirov ,
Thank you for the explanation. Actually, I've changed in the Vue templates the links to `res://menu`, but I still receive the same error with `res://navigation/menu` while I try to run it in Preview via `tns preview`. Should I somehow force to recompile the application?

**Comment by pavel-arapov** [comment]
Actually, the Preview app shows me the same error and doesn't load any new application.

**Comment by rosen-vladimirov** [comment]
Hey @pavel-arapov ,
Can you try keeping the Preview application open (i.e. do not close it when it fails with the `navigation/menu` error, instead just put it in background), open the Playground app and scan a QR code of another app? This should do the trick and fix the Preview app on your side.

**Comment by pavel-arapov** [comment]
It doesn't help me. I scan another app it opens me the error as it is without updating the preview. If I close the Preview and I scan the QR code, it tries to run the app, there is a crash of Preview app, and then I re-open I see the same error.

**Comment by rosen-vladimirov** [comment]
Hey @pavel-arapov , I've tried your steps and I was able to successfully recover the Preview application on my side. The important part here is to keep the preview application opened - in case you close it, you'll still receive the same error no matter what kind of project you are trying to scan.
In case you are unable to fix the problem, you can try cleaning the cache of the application, which AFAIK on iOS means uninstall and install of the application.

Hope this helps.

**Comment by pavel-arapov** [comment]
Hi @rosen-vladimirov ,
I reinstalled (already few times) the Preview App and I cannot run any application, I can say that I have my initial problem again. I see the default Preview screen with the list of elements. During the first run after the installation when I try to run the app, I see "Configuring Live Sync", but nothing is happening after.  No idea what I can do.

**Comment by pavel-arapov** [comment]
I don't know if it helps, but the only message I have in `tns preview --log trace` after scanning the QR code is `Received onLogSdkMessage message:  w5D2miW0N Exception: didn't receive device connected message after 11 retries`

**Comment by pavel-arapov** [comment]
Btw, I play with `tns run ios` and it works only with `res://navigation/menu` and it has an error on `res://menu` reference. Probably, I misunderstand something, but the resources are in `App_Resources` and the `menu.png` exists in the `navigation` folder. Why does the Preview App use a different structure? I use the following template for the tests `https://market.nativescript.org/plugins/tns-template-drawer-navigation-vue` without any modification. 
XCode: 11.1.0 
CocoaPods: 1.8.4
NativeScript CLI: 6.2.0

**Comment by rosen-vladimirov** [comment]
Hey @pavel-arapov ,
Regarding the error `Received onLogSdkMessage message: w5D2miW0N Exception: didn't receive device connected message after 11 retries` - this may happen when device does not have access to internet or the connection is flaky. The Preview application uses PubNub for communication, so you must have access to it for the communication to work.
Regarding the issue with `tns run`, please excuse me for not describing the problem clearly. When you create a new application with `tns create ...` you use some of our templates. When you use `tns preview` the local files are uploaded in the Preview application on device and the changes are synced in it. However, some of the files cannot be synced - all resources under `App_Resources` directory for example, i.e. all files used with `res://` must be prebuilt in the Preview application in order to make them work.
When you use `tns run` the application is built locally on your machine and all files, including the `App_Resources` are included in the built application, so you can reference you local images with `res://`.
So, what is the actual problem you've hit - in the latest version of the Preview application, we've moved some of the files in App_Resources (particularly the `menu` images for iOS), but we did not move them in the templates. When you use one of the templates where the `menu` images are used and try `tns preview` for iOS, the application fails as the code of the template expects them to be in a location, which is not available in the Preview app. After applying the change I've mentioned (replacing `res://navigation/menu` with `res://menu`, you can use the `tns preview`, but application fails with `tns run ios` as your local images in App_Resources are still under `App_Resources/iOS/navigation/...`. 
To resolve the second issue and use both `tns preview` and `tns run`, please move the images from `App_Resources/iOS/navigation/` to `App_Resources/iOS/`. This way both commands should work correctly.

We'll apply a fix in our templates, so newly created projects will work correctly. Thanks for reporting this problem.

**Comment by EdJones** [comment]
@rosen-vladimirov, thanks for working this! I've been fighting this all weekend, too. 

I'd come up with a workaround to put the icon in  `icon="~/images/navigation/menu"`
But of course, that's not a great solution across my 20 pages, let alone across all developers. 

Testing your fix, copying the icons to App_Resources/iOS/ did not work. 

If I also change the ActionItem link to
` icon="res://menu"`
that does work.

But, again, I don't wish to change all of my 20 pages, and I'd hate to own a much larger app.

**Comment by EdJones** [comment]
@rosen-vladimirov, I'm glad I didn't switch all those pages.
When I built the app through SideKick and downloaded it to my iPhone, the page with `icon="res://menu"`
crashes:
`CONSOLE ERROR file:///node_modules\@nativescript\core\trace\trace.js:174:30 Error: Could not load action bar icon: res://menu`

However, the page with
` icon="~/images/navigation/menu"`
works.

**Comment by rosen-vladimirov** [comment]
Hey guys,

I would like to summarize the information in this thread:
All resources used in your application with `res://` must be prebuilt, i.e. they **must** be part of the build process of the application itself. The path after `res://` is the path to the resource you want to use. For example, `res://navigation/menu` means that you **must** have a file called `menu.png` inside `navigation` directory in all of your `<path to App_Resources>/Android/drawable-<resolution>/` directorties (`<path to App_Resources/iOS` for iOS). This resource must exist when the application is build.

 There are two commands discussed in the thread:
 - `tns run ...` - when using this command, the application is built on your local machines and the resources are built exactly in the way they are in your App_Resources directory
- `tns preview` - when using this command application is not built at all and none of the resources in the App_Resources directory are part of it. Your local code is included in an already built application (NativeScript Preview), which has its own App_Resources structure.

The issue, discussed in this thread, was caused by a discrepancy between the structure of the App_Resources in our NativeScript Preview application (the one used with `tns preview`) and the templates used for creating the application. This led to a problem that some applications, created with a specific version of our templates, could not be used with `tns preview` out of the box.
We've already fixed this discrepancy, so I'm closing this issue.
